### ISSUE TITLE
Does the IgnoreJavaVersion flag still work? 

### ISSUE THREAD

**Issue by canthisbe** [PaperSpigot/Paper#9500]
### Stack trace

```
[Pre-Init] PojavLauncher INIT!
[Pre-Init] Version: 2.2-development
[Pre-Init] Commit: d9877bc (main)
[Pre-Init] Device: iPad Pro (12.9-inch)
[Pre-Init] iOS 16.5.1 (Unjailbroken)
[Pre-init] Entitlements availability:
* com.apple.developer.kernel.extended-virtual-addressing: NO
* com.apple.developer.kernel.increased-memory-limit: NO
* com.apple.private.security.no-sandbox: NO
* dynamic-codesigning: NO
[Debugging] Debug log enabled: YES
[Pre-init] Restored game directory preference (server)
[MCDL] Downloading purpur-1.20.1-2016
[MCDL] Begin: download libraries
[MCDL] Skipped library purpur-1.20.1-2016.jar
[MCDL] Finished: download libraries
[MCDL] Begin: download assets
[MCDL] Finished: download assets
[JavaLauncher] Beginning JVM launch
[JavaLauncher] Attempting to change to Java 17 (actual might be higher)
[JavaLauncher] JAVA_HOME is now set to /var/containers/Bundle/Application/BD678CFB-F8C6-4555-917F-542A50DBF162/App.app/java_runtimes/java-17-openjdk
[JavaLauncher] Restored preference: RENDERER is set to auto
[JavaLauncher] Max RAM allocation is set to 3072 MB
[JavaLauncher] Added custom JVM flag: -DPaper.IgnoreJavaVersion=true
[Init] Found JLI lib
[Init] Calling JLI_Launch
WARNING: Unknown module: cpw.mods.bootstraplauncher specified to --add-exports
Arc Capes DNS Injector
Parts of Alibaba's DCM library were used, please read https://github.com/alibaba/java-dns-cache-manipulator/blob/main/README.md for more info
Added DNS cache entry: s.optifine.net/23.95.137.176
Launching Minecraft purpur-1.20.1-2016
Library net.java.dev.jna:jna:5.12.1 has been changed to version 5.13.0
Ignored non-exists file: /var/mobile/Containers/Data/Application/46E0A65F-1FE6-4C9B-8F9B-9703359EC49F/Documents/Library/Application Support/minecraft/libraries/io/netty/netty-transport-native-epoll/4.1.82.Final/netty-transport-native-epoll-4.1.82.Final.jar
Args init finished. Now starting game
Starting org.bukkit.craftbukkit.Main
Unsupported Java detected (17.0.8-internal). You are running an outdated, pre-release version. Only general availability versions of Java are supported. Please update your Java version.
exit(0) called
```

### Plugin and Datapack List

Inapplicable

### Actions to reproduce (if known)

Directions for iOS, cannot confirm if Android is similar
1. [Install PojavLauncher](https://pojavlauncherteam.github.io/INSTALL.html)
2. Go to Files > On My iDevice > PojavLauncher > instances > default
3. Create folder named 'versions' (if not present)
4. Inside create another folder named exactly as the server jar (i.e. 'paper-1.20.1-82')
5. Place jar inside that folder
6. Place json named exactly as the jar with this: [paper-1.20.1-82.txt](https://github.com/PaperMC/Paper/files/12063505/paper-1.20.1-82.txt) (rename file ext, should work w/o changes if using paper-1.20.1-82, if not change first occurrence of sha1 and size to sha1 hash and size of jar in bytes respectively, rename id and filename to name of jar)
7. Open PojavLauncher, go to Settings > Java Tweaks > add flag -DPaper.IgnoreJavaVersion=true in JVM Launcher arguments, press play
8. Check results in Files > On My iDevice > PojavLauncher > latestlog.txt

### Paper version

/version is unavailable, though jar was latest download from website: Paper 1.20.1, Build 82, b05e70b3b3e1118b64a308411f7202589431f5599014cdf2fabda8ba1808e9dd

### Other

I'm trying to get a Minecraft server running on an iPad using PojavLauncher by gaslighting it to thinking its running a client. I've done this by mimicking the same folder structure of .minecraft and the <verison>.json, derived from 1.20.1.json. Log suggests it was successful at running the jar, but Paper mistakes the internal java Pojav uses as outdated. I've tried adding the flag `-DPaper.IgnoreJavaVersion=true` mentioned at #7829, but it made no difference.

**Comment by lynxplay** [comment]
The IgnoreJavaVersion flag was generally removed in regards to the actual java version of the jdk/jre and just allows anything.
You are hitting a different check that prevents unofficial builds of java from running, which does not have a flag. (See #9471, same issue).
Generally there was some discussion around making that check optional via a flag, however not really the highest priority as you are obviously running in a very cursed environment :sweat_smile:

**Comment by Warriorrrr** [comment]
You should be able to workaround this issue by using a java version higher than 17, since the check added by spigot doesn't cover those.

**Comment by lynxplay** [comment]
Resolved by #9692.
